### ISSUE TITLE
Issue with microsoft package on nuget restore of project.

### ISSUE THREAD

**Issue by dazinator** [Microsoft/VSProjectSystem#181]
I recently checked out my project system project on a new laptop and did a nugget restore of all the packages.

This is what happened:

Output window:

```
Successfully uninstalled 'stdole 7.0.3301' from DnnProjectSystem.ProjectType
Executing script file 'C:\Users\darre\Source\Repos\DnnProjectSystem\src\packages\Microsoft.VisualStudio.Threading.Analyzers.15.0.240\tools\uninstall.ps1'
Error HRESULT E_FAIL has been returned from a call to a COM component.
At C:\Users\darre\Source\Repos\DnnProjectSystem\src\packages\Microsoft.VisualStudio.Threading.Analyzers.15.0.240\tools\uninstall.ps1:14 char:17
+ ...             $project.Object.AnalyzerReferences.Remove($analyzerFilePa ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : OperationStopped: (:) [], COMException
    + FullyQualifiedErrorId : System.Runtime.InteropServices.COMException
 
Removed package 'Microsoft.VisualStudio.Threading.Analyzers 15.0.240' from 'packages.config'
Successfully uninstalled 'Microsoft.VisualStudio.Threading.Analyzers 15.0.240' from DnnProjectSystem.ProjectType
Removed package 'Microsoft.VisualStudio.Shell.15.0 15.0.26201' from 'packages.config'

```

This same error happens everytime I attempt a restore.

If I look at the state of the project references afterwards, most of them are left in an invalid state:

![image](https://cloud.githubusercontent.com/assets/3176632/22953992/d5848142-f30b-11e6-8731-9568b5f905d2.png)

I have tried doing an `Update-Package --reinstall` but the same problem happens every time..

From the output it looks as though there is a problem with the install.ps1 script for one of the nugget packages: Microsoft.VisualStudio.Threading.Analyzers.15.0.240\tools\uninstall.ps1:14 

So I uninstalled that particular package:

```

Executing script file 'C:\Users\darre\Source\Repos\DnnProjectSystem\src\packages\Microsoft.VisualStudio.Threading.Analyzers.15.0.240\tools\uninstall.ps1'...
Error HRESULT E_FAIL has been returned from a call to a COM component.
At C:\Users\darre\Source\Repos\DnnProjectSystem\src\packages\Microsoft.VisualStudio.Threading.Analyzers.15.0.240\tools\
uninstall.ps1:14 char:17
+ ...             $project.Object.AnalyzerReferences.Remove($analyzerFilePa ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : OperationStopped: (:) [], COMException
    + FullyQualifiedErrorId : System.Runtime.InteropServices.COMException

```

Then tried a nuget restore again. This time the same thing happened again but with a different package:

```

Executing script file 'C:\Users\darre\Source\Repos\DnnProjectSystem\src\packages\Microsoft.VisualStudio.ProjectSystem.Analyzers.15.0.594-pre\tools\uninstall.ps1'
Error HRESULT E_FAIL has been returned from a call to a COM component.
At C:\Users\darre\Source\Repos\DnnProjectSystem\src\packages\Microsoft.VisualStudio.ProjectSystem.Analyzers.15.0.594-pre\tools\uninstall.ps1:14 char:17
+ ...             $project.Object.AnalyzerReferences.Remove($analyzerFilePa ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : OperationStopped: (:) [], COMException
    + FullyQualifiedErrorId : System.Runtime.InteropServices.COMException

````

**Comment by adrianvmsft** [comment]
The 2 scripts that fail are responsible with adding and removing the analyzer references to/from the project.
As a workaround you can manually remove these 2 analyzers from the project. They are not needed for building or running your code; they are only used at design time for detecting potential issues in the code. You can add them back later via nuget package manager.
To manually remove them:
- remove the corresponding entries from `packages.config`
- manually edit the project file and remove the corresponding Analyzer references
  ```xml
  <ItemGroup>
    <Analyzer Include="..\..\packages\Microsoft.VisualStudio.ProjectSystem.Analyzers.15.0.716-pre-gfda68d7302\analyzers\Microsoft.VisualStudio.ProjectSystem.Analyzers.dll" />
    <Analyzer Include="..\..\packages\Microsoft.VisualStudio.Threading.Analyzers.15.0.20-pre\analyzers\Microsoft.VisualStudio.Threading.Analyzers.dll" />
  </ItemGroup>
  ```
However, it puzzles me why the other references are displayed with the warning icon (EnvDTE, etc). Did you notice any other changes to the project? Does that work after removing the analyzers? If those references still fail, it may indicate a larger problem.

Meantime, we are working on a new version Project System Extensibility, so we will also investigate this issue as part of that.

**Comment by dazinator** [comment]
Ok thank you, I will try what you have suggested.

I noticed that during the Nuget install process, all those references / packages actually first get installed successfully and the references look good. However, just before this problematic package is encountered, something seems to remove all the references from the project, and then then re-add them back but now, they are in the invalid state. My assumption was that a powershell install script is doing this, and as it's bombing out it doesn't get to clean up the references - but that is just a guess.

**Comment by adrianvmsft** [comment]
I've just tried using a new project type created with the last public version of Project System Extensibility, I updated the analyzers to the latest nuget packages, it all worked fine for me.
Could you try if you can get it to repro using a brand new project type?
There is a chance those scripts may not be needed anymore thanks for better nuget support in Visual Studio, I will see if I can get remove them.

**Comment by dazinator** [comment]
Weirdly I can only get this to repro on my laptop! I have simply removed the packages for now.
I am going to be setting up an AppVeyor build soon, so I'll see if the issue repro's there.

**Comment by dazinator** [comment]
I do still have a problem with this, but I am going to investigate a bit further to see:

1. Is it just with Extensibility project that this occurs, or all VS projects (i.e is it a wider issue with VS install).
2. Does it happen with a new vanilla project, or does it happen only after NuGet has been involved in a package restore operation.

Once I can narrow it down a little more, I'll hopefully know how to progress.

**Comment by dazinator** [comment]
Closing this, as works fine with a brand new project, so assuming my project has become corrupted.

**Comment by larrybud2004** [comment]
check our CSPROJ file for old references to the old version. Happens ALL-THE-TIME with certain packages in the csproj file.
